avcheck.py

```diff
@@ -0,0 +1,606 @@
+                return vendor_name
+        
+        return "Unknown AV Product"
+    
+    def _generator(self):
+        kernel_module_name = self.config["kernel"]
+        
+        # Collect all data
+        av_procs = self._detect_av_processes(kernel_module_name)
+        av_services = self._detect_av_services()
+        av_dlls = self._detect_av_dlls(kernel_module_name)
+        av_files = self._detect_av_files()
+        
+        # Get detected vendors
+        detected_vendors = set()
+        if av_procs:
+            for proc in av_procs:
+                vendor = self._identify_av_vendor(proc['name'], proc['path'])
+                detected_vendors.add(vendor)
+        
+        # EVERY yield must have exactly 4 values
+        yield (0, ("=" * 80, "", "", ""))
+        yield (0, ("ANTIVIRUS DETECTION RESULTS", "", "", ""))
+        yield (0, ("=" * 80, "", "", ""))
+        yield (0, ("", "", "", ""))
+        
+        # Detected Products
+        if detected_vendors:
+            yield (0, ("Detected AV Products:", "", "", ""))
+            for vendor in sorted(detected_vendors):
+                yield (0, (f"  * {vendor}", "", "", ""))
+            yield (0, ("", "", "", ""))
+        
+        # Running Processes
+        if av_procs:
+            yield (0, ("Running AV Processes:", "", "", ""))
+            for proc in av_procs:
+                vendor = self._identify_av_vendor(proc['name'], proc['path'])
+                yield (0, (
+                    f"  * {proc['name']}",
+                    f"PID: {proc['pid']} | PPID: {proc['ppid']}",
+                    f"Vendor: {vendor}",
+                    proc['path']
+                ))
+            yield (0, ("", "", "", ""))
+        else:
+            yield (0, ("Running AV Processes:", "", "", ""))
+            yield (0, ("  * No AV processes detected", "", "", ""))
+            yield (0, ("", "", "", ""))
+        
+        # Services
+        if av_services:
+            yield (0, ("AV Services:", "", "", ""))
+            for svc in av_services:
+                yield (0, (
+                    f"  * {svc['name']}",
+                    f"State: {svc['state']}",
+                    f"Display: {svc['display']}",
+                    svc['binary']
+                ))
+            yield (0, ("", "", "", ""))
+        
+        # DLLs (limited output)
+        if av_dlls:
+            yield (0, ("AV DLLs (Top 15):", "", "", ""))
+            for dll in av_dlls[:15]:
+                yield (0, (
+                    f"  * {dll['name']}",
+                    f"Process: {dll['process']} (PID: {dll['pid']})",
+                    f"Base: {dll['base']}",
+                    dll['path']
+                ))
+            if len(av_dlls) > 15:
+                yield (0, (f"  ... and {len(av_dlls) - 15} more DLLs", "", "", ""))
+            yield (0, ("", "", "", ""))
+        
+        # Detection Files (categorized) - ALWAYS SHOW IF THERE ARE FILES
+        if av_files:

```

Comment:
Almost everything in the following section is the wrong way of yielding data in volatility.  Please provide data and allow the UI (and therefore ultimately the user) the ability to sort and group the results.